applications/ContactStructuralMechanicsApplication/custom_utilities/contact_utilities.h

```diff
@@ -981,19 +946,50 @@ class ContactUtilities
      * @return var_matrix: The matrix containing the variables of the geometry
      */
     
+    static inline Matrix GetVariableMatrix(
+        const GeometryType& Nodes,
+        const Variable<array_1d<double,3> >& rVarName,
+        const unsigned int& Step
+        )
+    {
+        /* DEFINITIONS */        
+        const std::size_t num_nodes = Nodes.size();
+        const std::size_t dim = Nodes.WorkingSpaceDimension();
+        Matrix var_matrix(num_nodes, dim);
+        
+        for (unsigned int i_node = 0; i_node < num_nodes; i_node++)
+        {
+            const array_1d<double, 3> value = Nodes[i_node].FastGetSolutionStepValue(rVarName, Step);
+            for (unsigned int i_dof = 0; i_dof < dim; i_dof++)
+            {
+                var_matrix(i_node, i_dof) = value[i_dof];
+            }
+        }
+        
+        return var_matrix;
+    }
+    
+    /**
+     * It calculates the matrix of a variable of a geometry
+     * @param Nodes: The geometry to calculate
+     * @param rVarName: The name of the variable to calculate
+     * @param step: The step where calculate
+     * @return var_matrix: The matrix containing the variables of the geometry
+     */
+    
     template< unsigned int TDim, unsigned int TNumNodes>
     static inline Matrix GetVariableMatrix(
         const GeometryType& Nodes,
         const Variable<array_1d<double,3> >& rVarName,
-        const unsigned int step
+        const unsigned int& Step
```

Comment:
It is not needed to be a reference. Copying is less complex with same result

Reply:
Ok